plugin/adsk/scripts/mayaUsd_pluginUICreation.mel

```diff
@@ -2,9 +2,16 @@ int $gHasPrimUpdater = -1;
 
 global proc int hasPrimUpdater()
 {
-    global int $gHasPrimUpdater;
+    global int $gHasPrimUpdater = -1;
     if ($gHasPrimUpdater < 0) {
-        $gHasPrimUpdater = `python("def checkPrimUp():\n  try:\n    from mayaUsd.lib import PrimUpdaterManager;\n    return 1;\n  except:\n    return 0;\ncheckPrimUp()")`;
+        // Note: we use a convoluted call to exec() in Python because
+        //       try block in Python cannot be one-liners and the MEL
+        //       python() function only returns the computed value when
+        //       the supplied Python code is a one-liner.
+        //
+        //       Putting the Python try block inside exec() allow us to
+        //       makethe Python code look like a one-liner to MEL.
+        $gHasPrimUpdater = `python("res = -1; exec('try:\\n    from mayaUsd.lib import PrimUpdaterManager;\\n    res = 1;\\nexcept:\\n    res = 0;\\n'); res")`;

```

Comment:
Note: exec() uses the locals from the caller, so the res set within exec(0 is seen in the res variable outside it.

Reply:
In python 2.7, exec is a statement whereas in py3 its a function. Is this code expected to work in Maya 2022 when using py2?

Reply:
But the calling convention is the same, isn't it? What would break?

Reply:
I just tried it in a python 2.7 prompt and the code worked.

Reply:
`>>> res = -1; exec('try:\n    from mayaUsd.lib import sys;\n    res = 1;\nexcept:\n    res = 0;\n'); print(res)
0
>>> res = -1; exec('try:\n    import sys;\n    res = 1;\nexcept:\n    res = 0;\n'); print(res)
1
>>>
`

Reply:
Excellent. If it works, then I have no concerns.